api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -595,6 +595,16 @@ message Cluster {
 
     // Common Configuration for all consistent hashing load balancers (MaglevLb, RingHashLb, etc.)
     ConsistentHashingLbConfig consistent_hashing_lb_config = 7;
+
+    // Expected override host statuses. The result of load balancing is allowed to be overwritten
+    // only when the target host status is as expected when stateful session stickiness directly
+    // specifies a target host.
+    //
+    // If this is set and and has no any item then any host from stateful session stickiness will be
+    // ignored.
+    // If this is not set then healthy statuses (``UNKNOWN``, ``HEALTHY`` and ``DEGRADED`` for now)
+    // will be applied by default.
```

Comment:
Is this field only relevant when the stateful_session extension is used?
If so, can you add this to the comment, and also add a reference to it?

Reply:
Should we be referring to the specific extension here at all? Or should we instead have this talk about the host override feature in general? I can imagine other extensions wanting to use host overrides as well

Reply:
For the moment, it does. I will add some more comments. Thanks.

Reply:
One thing that I do not fully grok is what is expected if a config has this field set, but doesn't have the sticky_session extension configured.

Reply:
> Should we be referring to the specific extension here at all? Or should we instead have this talk about the host override feature in general? I can imagine other extensions wanting to use host overrides as well

@htuch This configuration is designed for general host override. But `LoadBalancerContext::overrideHostToSelect()` is internaly C++ API which is annomymous for public users and the stateful session is the only related implemenatation for now.
May be we can ref the specific extension just as an example like:

```
The result of load balancing is allowed to be overwritten by the L4/L7 filters specified host.
For example the `stateful session filter <........>` will specfic upstream host directly accoriding to the 
downstream request.
But the overridden will only worked when the status of specficed host is as expected in the configuration.

```

Reply:
I'm a bit lost reading this. I think there is context assumed of the reader that isn't likely to be there in most cases. Can you explain this section as if the reader had never heard of this override feature, pointing them to the relevant docs with RST links and/or some examples?

Reply:
The override host feature is currently only available through the L4/L7 extensions. Now stateful session is the only extension that is ready.

So my intention is to use the stateful session as an example to help the reader understand the functionality and the configuration.

I will think about how to make it clearer and easier to understand. 🤔